### ISSUE TITLE
Hardware configuration for this

### ISSUE THREAD

**Issue by deeprun** [jcjohnson/fast-neural-style#69]
Could anyone recommend some hardware configurations (CPU, GPU, memory, disk, etc) for fast-neural-style training? I am looking for system that can complete a full COCO dataset 40000 iteration training whthin about 3-4 hours. 

TH

**Comment by jcjohnson** [comment]
The only thing that really matters is GPU. To finish training in 3-4 hours
you'll need a Pascal Titan X or GTX 1080.

**Comment by deeprun** [comment]
Are these are from Nvidia, right?

**Comment by jcjohnson** [comment]
Yes. Nothing from AMD currently matches NVIDIA's deep learning performance.

**Comment by bgon** [comment]
Hi!
My hardware configuration with Ubuntu 16.04 OS
- RAM: 8GB
- CPU: Intel® Core™ i3-2105 CPU @ 3.10GHz × 4
- GPU: GeForce GTX 750 Ti 2GB

The specs form the GPU are on the low-end compared to a Pascal Titan X or a GTX 1080, but it takes 4h30m with a full COCO dataset and 40000 iterations, I have to use `-batch_size 1` to avoid memory issues on the GPU (I'm not sure how this option affect the end result..):
```
th train.lua \
  -h5_file data/ms-coco-full.h5 \
  -style_image images/styles/leaf.jpg \
  -style_image_size 384 \
  -content_weights 1.0 \
  -style_weights 5.0 \
  -checkpoint_name leaf \
  -batch_size 1 \
  -gpu 0
```

**Comment by htoyryla** [comment]
I have GTX1070, it has less CPU cores than 1080 and a bit slower memory bus, but if one is on a budget then it is also an option to consider. I have not measured the time for 40k iterations, but I am running several trainings per day. Here the 1070 was around 500 euros while 1080 was 800 euros.

**Comment by deeprun** [comment]
Thank you. Your numbers are very impressive. I rent a AWS g2.2xlarge and it took 10-12 hours to complete one training. And it was very expensive. It costed me more that $100 for one week. So I decided to build my own machine.

**Comment by deeprun** [comment]
Just want to confirm this is the correct Titan X card: http://www.geforce.com/hardware/10series/titan-x-pascal

**Comment by jcjohnson** [comment]
Yep that's the right one, but keep this in mind:

Pascal Titan X is the single fastest card you can buy right now, but if you
are concerned about money then the GTX 1080 is a much better value; Titan X
is $1200 compared to $600-$700 for a GTX 1080, and Titan X is only about
35% faster than a Titan X (https://github.com/jcjohnson/cnn-benchmarks).

The 1080 only has 8GB of memory but that should be enough for
fast-neural-style training with the default model architecture and batch
size; in fact many of my models were trained on a Titan Z which has only
6GB memory.
